internal/pkg/instrumentation/bpf/google.golang.org/grpc/server/bpf/probe.bpf.c

```diff
@@ -167,3 +170,28 @@ int uprobe_http2Server_operateHeader(struct pt_regs *ctx)
 
     return 0;
 }
+
+// func (ht *http2Server) WriteStatus(s *Stream, st *status.Status)
+// https://github.com/grpc/grpc-go/blob/bcf9171a20e44ed81a6eb152e3ca9e35b2c02c5d/internal/transport/http2_server.go#L1049
+SEC("uprobe/http2Server_WriteStatus")
+int uprobe_http2Server_WriteStatus(struct pt_regs *ctx) {
+    struct go_iface go_context = {0};
+    get_Go_context(ctx, 2, stream_ctx_pos, true, &go_context);
+    void *key = get_consistent_key(ctx, go_context.data);
+
+    struct grpc_request_t *grpcReq_event_ptr = bpf_map_lookup_elem(&grpc_events, &key);
+    // if grpcReq_event is null, then handleStream probe didn't run. Try starting a new span here
+    if (grpcReq_event_ptr == NULL)
+    {
+        bpf_printk("failed to get grpcReq_event from events map");
+        return -1;
+    }
+
+    void *status_ptr = get_argument(ctx, 3);
+    void *s_ptr = 0;
+    bpf_probe_read_user(&s_ptr, sizeof(s_ptr), (void *)(status_ptr + status_s_pos));
+    // Get status code from Status.s pointer
+    bpf_probe_read_user(&grpcReq_event_ptr->status_code, sizeof(grpcReq_event_ptr->status_code), (void *)(s_ptr + status_code_pos));
+    bpf_map_update_elem(&grpc_events, &key, grpcReq_event_ptr, 0);
```

Comment:
nit: I think this update call is redundant since we have a pointer to the map element and we are updating the underlying memory, but worth making sure.

Reply:
Didn't know that, thought it was a copy. Removed this line and it works thanks